### DISCUSSION TITLE
subtitles too long

### DISCUSSION THREAD

**Discussion by livos** [absadiki/subsai#69] [General]
Great project! Thank you. I just wonder why the subtitles are so long? They often show on 3 lines when generally subtitles show only on 2 lines max most part of the time. It takes too much space on the picture. Or maybe there is a way to set this ?

**Comment by absadiki** [comment]
You are welcome @livos :)

This depends on the model you are using. Some models provide a way to limit the length of each sentence, and some not.
For example you can choose `ggerganov/whisper.cpp` from the models, then from the `configs` you'll need to set `token_timestamps` to `True` and `max_len` to the maximum number of characters you want, this way you can control the maximum length of each subtitle. 

Hope this helps ?

**Comment by livos** [comment]
Ok. I understand now. Thank you!

**Comment by raivisdejus** [comment]
I am using HuggingFace model to load a fine tuned whisper, everything in the tool works great, very useful and easy to use.

Generated subtitles show up word by word, I think it would be useful if we could combine the words into longer sections, with ability to specify some max length of the combined subtitle section. This could be a post processing tool on top of whatever whisper gives us.

**Reply by absadiki** [reply]
Good idea @raivisdejus! 
so basically take the first segment and add the subsequent ones until you reach some max length ? right ?

**Reply by raivisdejus** [reply]
@abdeladim-s Best solution in real life is yet to be determined by testing. I did a proof of concept version of such "subtitle combiner" here https://gist.github.com/raivisdejus/07ca2e37d1fb87f81df12e424cf9175b (different setup, but similar approach)
This will combine words until it sees end of sentence punctuation `.`, `?` or `!`. Max word limit or max subtitle segment length time wise are also options. I chose the end of sentence punctuation, for simplicity.

**Reply by absadiki** [reply]
Yes, a lot of approaches can be implemented. 
Well, combining until punctuation is also good, if you can add this function as a post processing tool and submit a PR I'll be happy to merge it, otherwise whenever I find some free time I'll try to do it.
